### PR TITLE
Feature/user activity

### PR DESCRIPTION
**Pull request by GregBaggings**
Added the initial UserActivity and UserDetailsActivity to track and change user details via sharedPreferences

### CODE REVIEW

File: app/build.gradle [hunk 1]
```diff
@@ -24,9 +24,13 @@ dependencies {
...
     implementation "com.android.support.constraint:constraint-layout:$rootProject.ext.constraintLayoutVersion"
+    implementation 'com.android.support.constraint:constraint-layout:1.0.2'
+    implementation 'com.android.support:support-v4:27.0.2'
     testImplementation "junit:junit:$rootProject.ext.junitVersion"
     androidTestImplementation "com.android.support.test:runner:$rootProject.ext.runnerVersion"
     androidTestImplementation "com.android.support.test.espresso:espresso-core:$rootProject.ext.espressoVersion"
     // maps
     implementation "com.google.android.gms:play-services-maps:$rootProject.ext.playServicesVersion"
+
+    compile 'com.android.support:preference-v7:27.0.2'
```
Comment (Arcfej) [inline app/build.gradle:35]: Use com.android.support:preference-v7:$rootProject.ext.supportLibVersion
This way all the modules will use the same dependency.

**Comment by vokod** [comment]
Hi! 
I only found some minor issues, but 14 of them :)
You frequently use hardcoded strings in textview.setText methods like this:
`tv_BloodType.setText("Blood Type: " + value);`
But it cannot be translated nor easily updated.
I suggest to extract these strings to xml and apply string resource formatting.
Here is an explanation:
http://envyandroid.com/android-string-xml-resource-formatting/

 As all the user-entered preferences are saved as strings, there have to be some value validation. In this state anything can be entered for age and weight, not only numbers. But I suppose this is left out for a later enhancement.

**Comment by GregBaggings** [comment]
Yes, the inputs are strings and considered for refactor later.

I'll extract the strings to the xmls and update the PR soon.

**Review by Arcfej — COMMENTED** [review]
I've just run through the code. I think I'll find bugs or suggestions only after the merge.

**Comment by vokod** [comment]
Then we can pull it up now, and later implement/reformat the things mentioned above.

**Comment by Arcfej** [comment]
Oh!
And please comment the code and/or write documentation. It's make the teamwork much easier.

File: app/build.gradle (hunk 1 above)
Comment (GregBaggings) [reply app/build.gradle:35]: Thanks, I'll change that.

**Comment by GregBaggings** [comment]
Sure thing, please wait with the merge until further notice. I'll add and change the stuff and check back when its done.

**Pull request closed without merging**